bootstrapper/internal/journald/journald.go

```diff
@@ -31,17 +31,17 @@ func NewCollector(ctx context.Context) (*Collector, error) {
 	if cmd.Err != nil {
 		return nil, cmd.Err
 	}
-	return &Collector{cmd}, nil
+	pipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	return &Collector{cmd, pipe}, nil
 }
 
-// Collect gets all journald logs from a service and returns a byte slice with the plain text logs.
-func (c *Collector) Collect() ([]byte, error) {
-	out, err := c.cmd.Output()
-	var exitErr *exec.ExitError
-	if errors.As(err, &exitErr) {
-		return nil, fmt.Errorf("executing %q failed: %s", c.cmd, exitErr.Stderr)
-	} else if err != nil {
-		return nil, fmt.Errorf("executing command: %w", err)
+// Pipe returns a pipe to read the systemd logs. This should be read with a bufio Reader.
+func (c *Collector) Pipe() (io.ReadCloser, error) {
```

Comment:
We need to add methods to get error and stdErr of the command in case of a failure.